File: java/src/main/java/io/fivexer/sdk/model/WorkflowDefinitionSummary.java
package io.fivexer.sdk.model;

/**
 * A workflow definition as it appears in a listing.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowDefinitionSummary {
    private String id;
    private String name;

    public String getId() { return id; }
    public String getName() { return name; }
}

File: php/src/Model/WorkflowDefinitionSummary.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A workflow definition as it appears in a listing.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowDefinitionSummary
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            name: (string) ($data['name'] ?? ''),
        );
    }
}
